code/modules/mob/living/simple_animal/constructs.dm

```diff
@@ -268,3 +270,88 @@
 /mob/living/simple_animal/hostile/construct/harvester/hostile //actually hostile, will move around, hit things
 	AIStatus = AI_ON
 	environment_smash = 1 //only token destruction, don't smash the cult wall NO STOP
+
+
+////////////////Glow////////////////////
+/mob/living/simple_animal/hostile/construct/proc/updateglow()
+	overlays = 0
+	var/overlay_layer = LIGHTING_LAYER + 1
+	if(layer != MOB_LAYER)
+		overlay_layer=TURF_LAYER+0.2
+
+	overlays += image(icon,"glow-[icon_state]",overlay_layer)
+	set_light(1, 1, l_color = "#FFFFFF")
+
+
+///ui stuff
+
+/mob/living/simple_animal/hostile/construct/armoured/handle_hud_icons_health()

```

Comment:
Look in data_huds.dm and see if you can make use of the `RoundDiagBar(value)` proc! It just returns text, so you can just use it for your HUD bar too, insteading of making a new proc!